File: mama/c_cpp/src/gunittest/cpp/MamaRegexChannelFilterTest.cpp
#include <gtest/gtest.h>
#include "mama/mama.h"
#include "wombat/property.h"
#include "mama/MamaRegexChannelFilter.h"
#include <cstring>

using namespace Wombat;

class MamaRegexChannelFilterTestCPP : public ::testing::Test
{	
protected:
    MamaRegexChannelFilterTestCPP () {};
    ~MamaRegexChannelFilterTestCPP () {};

    void SetUp() override {};
    void TearDown () override {};

};

TEST_F (MamaRegexChannelFilterTestCPP, CreateDestroy)
{
    // No crashing expected
    auto* filter = new MamaRegexChannelFilter;
    delete filter;
}

TEST_F (MamaRegexChannelFilterTestCPP, CheckDefaults)
{
    auto* filter = new MamaRegexChannelFilter;
    ASSERT_EQ (0, filter->getChannel("banana"));
    filter->setDefaultChannel(66);
    ASSERT_EQ (66, filter->getChannel("banana"));
    delete filter;
}

TEST_F (MamaRegexChannelFilterTestCPP, CheckFilteringIncludingMiss)
{
    auto* filter = new MamaRegexChannelFilter;
    filter->addRegex ("^[A-Z]", 1);
    filter->addRegex ("^[a-z]", 2);
    filter->addRegex ("^[0-9]", 3);
    filter->addRegex ("^[0-9]", 3);
    filter->setDefaultChannel (66);
    ASSERT_EQ (2, filter->getChannel("banana"));
    ASSERT_EQ (1, filter->getChannel("BANANA"));
    ASSERT_EQ (3, filter->getChannel("6 bananas"));
    ASSERT_EQ (66, filter->getChannel("!bananas :("));
    delete filter;
}

File: mama/dotnet/src/nunittest/MamaRegexChannelFilterTest.cs
using NUnit.Framework;
using Wombat;

namespace NUnitTest
{
    public class MamaRegexChannelFilterTest
    {
        [Test]
        public void TestCheckDefaults() {
            MamaRegexChannelFilter filter = new MamaRegexChannelFilter();
            Assert.AreEqual(0, filter.getChannel("banana"));
            filter.setDefaultChannel(66);
            Assert.AreEqual(66, filter.getChannel("banana"));

        }

        [Test]
        public void TestCheckFilteringIncludingMiss() {
            MamaRegexChannelFilter filter = new MamaRegexChannelFilter();
            filter.addRegex ("^[A-Z]", 1);
            filter.addRegex ("^[a-z]", 2);
            filter.addRegex ("^[0-9]", 3);
            filter.addRegex ("^[0-9]", 3);
            filter.setDefaultChannel (66);
            Assert.AreEqual(2, filter.getChannel("banana"));
            Assert.AreEqual(1, filter.getChannel("BANANA"));
            Assert.AreEqual(3, filter.getChannel("6 bananas"));
            Assert.AreEqual(66, filter.getChannel("!bananas :("));
        }
    }
}